Tests/LexTests.cs

```diff
@@ -7,68 +8,99 @@ namespace Tests
-    {
-      Assert.Equal(CC.Lex("{"), new List<Token>() {new Token("{", OpenBrace)});
-    }
+    public void SingleTokenReturnsToken() =>
+      Assert.Equal(lexer.Lex("{"), new List<Token>() { new Token("{", OpenBrace) });
 
     [Fact]
-    public void WhitespacesAreNeverReturned()
-    {
-      Assert.Equal(CC.Lex(" {"), new List<Token>() {new Token("{", OpenBrace)});
-    }
+    public void WhitespacesAreNeverReturned() =>
+      Assert.Equal(lexer.Lex(" {"), new List<Token>() { new Token("{", OpenBrace) });
 
     [Fact]
-    public void KeywordsAreReturned()
-    {
-      Assert.Equal(CC.Lex("return;"), new List<Token>()
+    public void KeywordsAreReturned() =>
+      Assert.Equal(lexer.Lex("return;"), new List<Token>()
       {
         new Token("return", Keyword),
         new Token(";", Semicolon),
       });
-    }
 
     [Fact]
     public void LexReturn2()
     {
-      Assert.Equal(CC.Lex("int main() { return 2; }"),
-        new List<Token>()
+      Assert.Equal(lexer.Lex("int main() { return 2; }"),
+        new TokenList
         {
-          new Token("int", Keyword),
-          new Token("main", Identifier),
-          new Token("(", OpenParen),
-          new Token(")", CloseParen),
-          new Token("{", OpenBrace),
-          new Token("return", Keyword),
-          new Token("2", Integer),
-          new Token(";", Semicolon),
-          new Token("}", CloseBrace),
+          { "int", Keyword},
+          { "main", Identifier},
+          { "(", OpenParen},
+          { ")", CloseParen},
+          { "{", OpenBrace},
+          { "return", Keyword},
+          { "2", Integer},
+          { ";", Semicolon},
+          { "}", CloseBrace},
         });
     }
 
     [Fact]
-    public void LexReturn2MinimumSpaces()
+    public void LexReturn2AndComments()
     {
-      Assert.Equal(CC.Lex("int main(){return 2;}"),
-        new List<Token>()
+      Assert.Equal(lexer.Lex("/*1*/int/*2*/main(/*3*/)/*4*/{/*5*/ return 2; }"),
+        new TokenList
         {
-          new Token("int", Keyword),
-          new Token("main", Identifier),
-          new Token("(", OpenParen),
-          new Token(")", CloseParen),
-          new Token("{", OpenBrace),
-          new Token("return", Keyword),
-          new Token("2", Integer),
-          new Token(";", Semicolon),
-          new Token("}", CloseBrace),
+          { "int", Keyword},
+          { "main", Identifier},
+          { "(", OpenParen},
+          { ")", CloseParen},
+          { "{", OpenBrace},
+          { "return", Keyword},
+          { "2", Integer},
+          { ";", Semicolon},
+          { "}", CloseBrace},
         });
     }
+
+    [Fact]
+    public void LexCommentNotening()
+    {
+      var ex = Assert.Throws<Exception>(() => lexer.Lex("/* return 2; }"));
+      Assert.Equal("unterminated comment", ex.Message);
+    }
+  
+    [Fact(Skip = "/* is allowed inside a comment")]
+    public void LexCommentInComment1()
```

Comment:
I'm not sure i understand this. Are comments in comments allowed? If thats the case, why are we expecting an Exception? And the mesage is strange as well "Unterminated comment"

Reply:
changed the test so it tests that "\*   \*   */" is okay